### PR TITLE
Convert to fstring

### PR DESCRIPTION
**Pull request by Behemkot**
https://github.com/pandas-dev/pandas/issues/29547

- [x] pandas/_libs/tslibs/frequencies.pyx
- [x] pandas/_libs/tslibs/period.pyx
- [x] pandas/_libs/tslibs/strptime.pyx

### CODE REVIEW

File: pandas/_libs/tslibs/frequencies.pyx [hunk 1]
```diff
@@ -223,7 +223,7 @@ cpdef _period_str_to_code(str freqstr):
     try:
         return _period_code_map[freqstr]
     except KeyError:
-        raise ValueError(INVALID_FREQ_ERR_MSG.format(freqstr))
+        raise ValueError(INVALID_FREQ_ERR_MSG(freqstr))
```
Comment (alimcmaster1) [inline pandas/_libs/tslibs/frequencies.pyx]: This looks like a typo?

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1209,8 +1209,8 @@ def period_format(int64_t value, int freq, object fmt=None):
         elif freq_group == 4000:  # WK
             left = period_asfreq(value, freq, 6000, 0)
             right = period_asfreq(value, freq, 6000, 1)
-            return '%s/%s' % (period_format(left, 6000),
-                              period_format(right, 6000))
+            return f'{period_format(left, 6000)}/{period_format(right, 6000)}'
```
Comment (alimcmaster1) [inline pandas/_libs/tslibs/period.pyx]: Double quotes for f string please

File: pandas/_libs/tslibs/period.pyx [hunk 3]
```diff
@@ -1598,7 +1599,7 @@ cdef class _Period:
     def __richcmp__(self, other, op):
         if is_period_object(other):
             if other.freq != self.freq:
-                msg = DIFFERENT_FREQ.format(cls=type(self).__name__,
+                msg = DIFFERENT_FREQ(cls=type(self).__name__,
```
Comment (alimcmaster1) [inline pandas/_libs/tslibs/period.pyx]: You need an f string?

**Review by alimcmaster1 — CHANGES_REQUESTED** [review]
few comments

File: pandas/_libs/tslibs/frequencies.pyx (hunk 1 above)
Comment (Behemkot) [reply pandas/_libs/tslibs/frequencies.pyx]: Hi,
In order to converting str.format() to f string I changed INVALID_FREQ_ERR_MSG definition from predefined string to lambda function. Now error message is defined in one place and is callable without .format().

Current definition:
INVALID_FREQ_ERR_MSG = lambda arg: f"Invalid frequency {arg}"

So now message is called as a function:
INVALID_FREQ_ERR_MSG(arg_str)

File: pandas/_libs/tslibs/period.pyx (hunk 2 above)
Comment (Behemkot) [reply pandas/_libs/tslibs/period.pyx]: single quotes changed to double quotes

File: pandas/_libs/tslibs/period.pyx (hunk 3 above)
Comment (Behemkot) [reply pandas/_libs/tslibs/period.pyx]: same deal as abowe in frequencies.pyx file. DIFFERENT_FREQ is now a lambda function

**Comment by Behemkot** [comment]
My changes are causing problems, closing pull request .

**Pull request closed without merging**